### PR TITLE
Add option to specify for roslyn server the razor output directory path

### PR DESCRIPTION
**Pull request by maryamariyan**
- Adds devKitRazorOutputPath option
- Allows RazorSourceGenerator to be picked up by tooling rather than SDK
- Fixes WorkItem 1926900 (https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1926900)

/cc @CyrusNajmabadi @davidwengier @dibarbet 
Related to https://github.com/dotnet/roslyn/pull/72482

### CODE REVIEW

File: src/lsptoolshost/roslynLanguageServer.ts [hunk 1]
```diff
@@ -510,6 +510,8 @@ export class RoslynLanguageServer {
             args.push('--extension', extensionPath);
         }
 
+        args.push('--devKitRazorOutputPath', 'C:/CodeHub/vscode-csharp/.razor');
```
Comment (maryamariyan) [inline src/lsptoolshost/roslynLanguageServer.ts:513]: oops, I did not realize I am directly passing this.

I will fix this in a next commit

**Comment by maryamariyan** [comment]
Applied PR feedback. Tested changes again for VS code and confirmed the problem goes away when project is targeting 6.0 and using 6.0 SDK.

Before change:
image


After change:
image

**Review by davidwengier — APPROVED** [review]

**Comment by maryamariyan** [comment]
The roslyn server version in vscode-csharp is not updated to a version that contains the server side change yet causing the failure on the CI.

```
[stderr] Unrecognized command or argument '--razorSourceGenerator'.
Unrecognized command or argument '/mnt/vss/_work/1/s/.razor/Microsoft.CodeAnalysis.Razor.Compiler.dll'.
```

Thanks @dibarbet for clarifying this

**Review by dibarbet — APPROVED** [review]

**Pull request merged**